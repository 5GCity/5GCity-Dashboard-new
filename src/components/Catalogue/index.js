/**
 * Catalogue Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { rgba } from 'polished'
import { catalogueUtils } from 'utils'
import { Start, OtherComponent, Stop, Plus, Remove, VirtualSwitch } from 'components/Icons'

export default ({ ...props }) => (
  <SVGContainer
    {...props}
    onClick={() => !props.disabled && props.onClick(getProps(props))}>
    <Group>
    {props.version &&
      <Version x="6" y="16">{props.version}</Version>
    }
    {props.type === "start" &&
      <Start transform="translate(33,20)" />
    }
    {props.type === "stop" &&
      <Stop transform="translate(33,20)" />
    }
    {props.type === "VS" &&
      <VirtualSwitch transform="translate(33,20)" />
    }
    {props.type === "VNF" &&
      <OtherComponent transform="translate(33,20)" {...props} />
    }
    {props.name &&
      <Name>
        {catalogueUtils(props.name)}
      </Name>
    }
    <Rect />
    { !props.disabled &&
    <PlusCatalogue transform="translate(33,40)" />
    }
    { props.disabled &&
    <RemoveCatalogue transform="translate(33,40)" />
    }
    </Group>
  </SVGContainer>
)

const getProps = props => {
  const { name, version, type, colortext, circlefill } = props
  const object = {};
  object.name = name;
  object.type = type ? type.toLowerCase() : name.toLowerCase();
  if(version) {
    object.version = version;
  }
  if(colortext) {
    object.colortext = colortext;
  }
  if(circlefill) {
    object.circlefill = circlefill;
  }
  return object;
}

const Group = styled.g``

const PlusCatalogue = styled(Plus)`
  visibility: hidden;
`

const RemoveCatalogue = styled(Remove)`
  visibility: visible;
`

const Rect = styled.rect`
  fill: transparent;
  width: 100px;
  height: 106px;
  rx: 10%;
  stroke-width: 1.5;
  stroke-dasharray: 5;
  stroke: ${({theme}) => rgba(theme.secondaryColor, 0.3)};
`

const Version = styled.text`
font-family: ${({ theme }) => theme.fontFamily };
fill: ${({ theme }) => theme.secondaryColor };
font-size: 10px;
`

const Name = styled.text`
  text-anchor: middle;
  font-family: ${({ theme }) => theme.fontFamily };
  fill: ${({ theme }) => theme.secondaryColor };
  font-size: 12px;
`
const SVGContainer = styled.svg`
  width: 100px;
  height: 106px;
  cursor: pointer;
  margin: 8px 4px;
  &:hover {
    ${Rect} {
      fill: ${({theme}) => rgba(theme.secondaryColor, 0.1)};
    }
    ${PlusCatalogue} {
      visibility: visible;
    }
  }

  ${props => props.disabled && `
    cursor: not-allowed;
    `
  }
`
