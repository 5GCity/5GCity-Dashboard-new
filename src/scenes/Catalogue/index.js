/**
 * Catalogue Scene
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React, { Component } from 'react'
import Logic from './logic'
import styled from 'styled-components'

/* Components */
import PageTitle from 'components/PageTitle'

/* Containers */
import NavBar from 'containers/Navbar'
import ListNewNetworks from 'containers/Lists/ListNewNetworks'

class Catalogue extends Component {
  render () {
    return (
      <Wrapper>
        <NavBar />
        <PageTitle
          title={'Network Service Catalogue'}
        />
        <ListNewNetworks />
      </Wrapper>
    )
  }
}

export default Logic(Catalogue)

const Wrapper = styled.div`
  margin-left: 120px;
  padding: 0 24px;
`
