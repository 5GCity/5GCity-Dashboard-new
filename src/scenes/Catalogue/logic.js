/**
 * Catalogue Container Logic
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */

import { kea } from 'kea'
import { API_SDK } from 'config'
import { put, call } from 'redux-saga/effects'
import { Organizations } from './utils'
import axios from 'axios'
import { PropTypes } from 'prop-types'

export default kea({
  path: () => ['scenes', 'Catalogue'],

  /*
  connect: {
    props: [
      Logic, [
        'prop1'
      ]
    ],
    actions: [
      Logic, [
        'action as newActionName'
      ],
    ]
  },
  */

  actions: () => ({
    fetchOrganizations : () => ({}),
    setCatalogueOrganizations: (organizations) => ({ organizations }),
    changeCatalogueOrganization : (organization) => ({organization}),
  }),

  reducers: ({ actions }) => ({
    organizationsList: [null, PropTypes.array, {
      [actions.fetchOrganizations] : (state, payload) => null,
      [actions.setCatalogueOrganizations]: (state, payload) => Organizations(payload.organizations)
    }],
    selectCatalogueOrganization: ['all', PropTypes.string, {
      [actions.changeCatalogueOrganization]: (state, payload) => payload.organization
    }],
  }),

  start: function * () {
    const { fetchOrganizations } = this.actions

    yield put(fetchOrganizations())
  },

  takeLatest: ({ actions, workers }) => ({
    [actions.fetchOrganizations]: workers.fetchOrganizations,
  }),

  workers: {

    * fetchOrganizations () {
      const { setCatalogueOrganizations } = this.actions
      try {
        let responseResult = yield call(axios.get, `${API_SDK}/sdk/sliceManagement/slices`)
        const { data } = responseResult
        yield put(setCatalogueOrganizations(data))
      } catch (error) {
        yield put(setCatalogueOrganizations(null))
        console.error(`Error ${error}`)
      }
    },
  }

})
