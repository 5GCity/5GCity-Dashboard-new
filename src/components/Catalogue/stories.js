/**
 * Catalogue Component Stories
 * Please write a description or remove this line
 *
 * @author Guilherme Patriarca <gpatiarca@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions';

import Catalogue from './index'
import styled from 'styled-components'

const ExampleOne = () => (
  <Wrapper>
    <Catalogue
      name={"Start"}
      type={'start'}
      onClick={action('click')}
    />
    <Catalogue
      version={"3.0.1"}
      type={'VNF'}
      colortext="blue"
      circlefill="yellow"
      name={"other mec with long name"}
      onClick={action('click')}
    />
    <Catalogue
      version={"3.0.1"}
      type={'VNF'}
      colortext="whiteSmoke"
      circlefill="green"
      name={"other mec name"}
      onClick={action('click')}
      disabled
    />
    <Catalogue
      name={"Virtual Switch"}
      type={'VS'}
      onClick={action('click')}
    />
  </Wrapper>
)

storiesOf('Catalogue', module)
  .add('default', ExampleOne)

  const Wrapper = styled.div``
