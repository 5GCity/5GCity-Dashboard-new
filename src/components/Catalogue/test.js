/**
 * Catalogue Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Catalogue from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Catalogue />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
