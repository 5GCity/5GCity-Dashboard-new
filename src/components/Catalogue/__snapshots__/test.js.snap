// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  width: 100px;
  height: 106px;
  cursor: pointer;
  margin: 8px 4px;
}

.c0:hover .c1 {
  fill: rgba(137,151,159,0.1);
}

.c0:hover .c3 {
  visibility: visible;
}

.c2 {
  fill: transparent;
  width: 100px;
  height: 106px;
  rx: 10%;
  stroke-width: 1.5;
  stroke-dasharray: 5;
  stroke: rgba(137,151,159,0.3);
}

.c4 {
  visibility: hidden;
}

<svg
  className="c0"
  onClick={[Function]}
>
  <g
    className=""
  >
    <rect
      className="c1 c2"
    />
    <g
      className="c3 c4"
      transform="translate(33,40)"
    >
      <rect
        fill="white"
        height="5"
        rx="0"
        width="32"
        y="13"
      />
      <rect
        fill="white"
        height="5"
        ry="0"
        transform="rotate(90)"
        width="32"
        y="-18"
      />
    </g>
  </g>
</svg>
`;
